folder.go

```diff
@@ -35,6 +35,17 @@ func (c *Client) Folder(id int64) (*Folder, error) {
 	return folder, err
 }
 
+// Folder fetches and returns the Grafana folder whose UID it's passed.
+func (c *Client) FolderByUID(uid string) (*Folder, error) {
+	folder := &Folder{}
+	err := c.request("GET", fmt.Sprintf("/api/folders/id/%s", uid), nil, nil, folder)
```

Comment:
According to https://grafana.com/docs/grafana/latest/http_api/folder/#get-folder-by-uid, the URL would be
```suggestion
	err := c.request("GET", fmt.Sprintf("/api/folders/%s", uid), nil, nil, folder)
```